drivers/mfd/mfd_axp192.c

```diff
@@ -164,9 +189,11 @@ int mfd_axp192_gpio_func_get(const struct device *dev, uint8_t gpio, enum axp192
 		return -EINVAL;
 	}
 
-	ret = i2c_reg_read_byte_dt(&(config->i2c), gpio_reg_desc[gpio].reg, &reg_fnc);
-	if (ret != 0) {
-		return ret;
+	if (gpio < ARRAY_SIZE(gpio_reg_desc)) {
```

Comment:
Is it worth moving the special case for EXTEN here instead?
It can return immediately.

Reply:
I am not sure what you mean here?

Are you proposing adding a separate if for setting `*func` for the EXTEN case (5) ?